app/scripts/directives/resources.js

```diff
@@ -1,6 +1,75 @@
 'use strict';
 
 angular.module('openshiftConsole')
+  .directive('containerStatuses', function($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        containerStatuses: '=',
+        startDebugTerminal: '=?',
+        detailed: '=?'
+      },
+      templateUrl: 'views/_container-statuses.html',
+      link: function(scope) {
+        scope.hasDebugTerminal = angular.isFunction(scope.startDebugTerminal);
+
+        scope.podContainerStatuses = _.get(scope.containerStatuses, 'status.containerStatuses', []);
+        scope.podContainerInitStatuses = _.get(scope.containerStatuses, 'status.initContainerStatuses', []);
+
+        scope.hasContainerFailed = function (containerStatus) {
```

Comment:
Shouldn't need to make this a scope var if it's not used in the view. Suggest just using

```js
var isContainerFailed = $filter('isContainerFailed');
```

then below calling `isContainerFailed()`